velox/connectors/hive/TableHandle.h

```diff
@@ -30,15 +30,38 @@ class HiveColumnHandle : public ColumnHandle {
  public:
   enum class ColumnType { kPartitionKey, kRegular, kSynthesized };
 
+#ifdef VELOX_ENABLE_BACKWARD_COMPATIBILITY
   HiveColumnHandle(
       const std::string& name,
       ColumnType columnType,
       TypePtr dataType,
       std::vector<common::Subfield> requiredSubfields = {})
+      : HiveColumnHandle(
+            name,
+            columnType,
+            std::move(dataType),
+            nullptr,
+            std::move(requiredSubfields)) {}
+  g
+#endif
+
+  /// NOTE: 'dataType' is the column type in target write table. 'hiveType' is
+  /// converted type of the corresponding column in source table which might not
+  /// be the same type, and the table scan needs to do data coercion if needs.
+  /// The table writer also needs to respect the type difference when processing
+  /// input data such as bucket id calculation.
+  HiveColumnHandle(
```

Comment:
This change seems unrelated. It would be nice to move it to a separate PR and add a check for dataType == hiveType, since we currently do not support coersion.